resources/js/Shared/InputField.vue

```diff
@@ -0,0 +1,275 @@
+<script lang='ts'>
```

Comment:
I gotta optimize this component at some point before it starts being used in a bunch of files. Its not bad but it could be improved I bet

Reply:
yeah its not fully "TS" right now :) but its not bad, I figured I would get it here and after its pushed we can hack on it to make it better.